nuto/mechanics/solver/Solve.h

```diff
@@ -1,18 +1,32 @@
 #pragma once
 #include <Eigen/Core>
-#include "nuto/mechanics/dofs/GlobalDofVector.h"
-#include "nuto/mechanics/dofs/GlobalDofMatrixSparse.h"
+#include "nuto/mechanics/dofs/DofVector.h"
+#include "nuto/mechanics/dofs/DofMatrixSparse.h"
 #include "nuto/mechanics/constraints/Constraints.h"
 
 namespace NuTo
 {
 
-Eigen::VectorXd Solve(Eigen::SparseMatrix<double> A, Eigen::VectorXd b);
+DofVector<double> Solve(const DofMatrixSparse<double>& K, const DofVector<double>& f, Constraint::Constraints& bcs,
```

Comment:
What do you think about replacing `Constraints::Constraints& bcs` with `DofMatrixSparse<double> Cmat`? This allows calculating CMat once the constraints change. Or maybe overload the method accordingly.